CodeGeneration/Sources/SyntaxSupport/CompatibilityLayers.swift

```diff
@@ -0,0 +1,192 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Computes and caches information about properties and initializers that ought to be generated for the compatibility layer.
+public struct CompatibilityLayers /*: Sendable*/ {
```

Comment:
I’m a little surprised by the use of `Layers` (plural). Layers apply to me that there are multiple layers on top of each other where one shadows the other. But all the data structures in here are flat, so would `CompatibilityLayer` make more sense?

Reply:
I wasn't sure about how we meant the term "compatibility layer" to be used—is there one compatibility layer that covers the entire project, or are there many "compatibility layers", each covering one type or API? I took the latter interpretation, and named this `CompatibilityLayers` because it has knowledge of all of them, but I can rename it if I've gotten this wrong.

Reply:
I consider there to be a single compatibility layer for the entire project.

Reply:
Gotcha.